assignment/scripts/3-music-collection.js

```diff
@@ -1 +1,123 @@
 console.log('***** Music Collection *****')
+
+
+let collection = [];
+
+function addToCollection(title, artist, yearPublished, tracks){
+    let newAlbum = {
+        title: title ,
+        artist: artist ,
+        yearPublished: yearPublished,
+        tracks: tracks,
+    }
+    collection.push(newAlbum);
+    return collection;
+};
+
+console.log('--add to collection--');
+console.log('New album added:', addToCollection('Enema of the State', 'blink 182', '1999'));
+console.log('New album added:', addToCollection('Take Off Your Pants and Jacket', 'blink 182', '2001'));
+console.log('New album added:', addToCollection('The Colour and the Shape', ' Foo Fighters', '1997'));
+console.log('New album added:', addToCollection('By the Way', 'Red Hot Chili Peppers', '2002'));
+console.log('New album added:', addToCollection('Dookie', 'Green Day', '1994'));
+console.log('New album added:', addToCollection('Meteora', 'Linkin Park', '2003'));
+console.log('New album added:', addToCollection('Hot Fuss', 'The Killers', '2004'));
+
+console.log('My album collection:',collection);
+
+console.log('--SHOW COLLECTION FUNCTION--');
+
+// - Add a function named `showCollection`. This function should:
+//   - Take in an array parameter. (This allows it to be reused to show any collection, like the results from the find or search.)
+//   - Console.log the number of items in the array.
+//   - Loop over the array and console.log each album's information formatted like: `TITLE by ARTIST, published in YEAR`.
+
+function showCollection(albumArray){
+    console.log('My albums so far:',albumArray.length)
+    for (newAlbum of albumArray){
+        console.log(`${newAlbum.title} by ${newAlbum.artist} in ${newAlbum.yearPublished}`);
+    }
+    return albumArray;
+}
+console.log('About my collection:', showCollection(collection)); 
+
+console.log('--FIND BY ARTIST FUNCTION--');
+
+
+// - Add a function named `findByArtist`. This function should:
+//   - Take in `artist` (a string) parameter
+//   - Create an array to hold any results, empty to start
+//   - Loop through the `collection` and add any objects with a matching artist to the array.
+//   - Return the array with the matching results. If no results are found, return an empty array.
+
+  
+function findByArtist(artist){
+    let artistArray = []
+    for (let newAlbum of collection){
+        if (artist.toLowerCase() === newAlbum.artist.toLowerCase());
+        artistArray.push(newAlbum);
+        return artistArray;
```

Comment:
Nice touch making the search case insensitive, but unfortunately the rest of this needs a bit of work.

Since you put a semicolon after the if-condition, you cut it off from the rest of your code. Whether it passes or not doesn't matter. The later code will just run normally as if the if-condition weren't there.

Additionally, you have a return statement in your for-loop. As soon as you hit the for-loop, the function is done, it is going to return. With the if-condition cut off, that means you will always return the first item in collection, no matter what the artist is. Even if the if-condition worked, you would still only ever return the first matching album. What if there were two albums that matched the artist you wanted to find?

This is the implementation you were looking for:

```suggestion
        if (artist.toLowerCase() === newAlbum.artist.toLowerCase()) {
            artistArray.push(newAlbum);
        }
```

No return statement in the loop. Put the push inside the if block (and drop the semicolon) and your code works.